ts/features/messages/screens/MessagesSearchScreen.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Divider,
+  IOSpacingScale,
+  IOStyles,
+  SearchInput,
+  SearchInputRef,
+  VSpacer
+} from "@pagopa/io-app-design-system";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { FlatList, ListRenderItemInfo, Platform, View } from "react-native";
+import { useFocusEffect } from "@react-navigation/native";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import I18n from "../../../i18n";
+import { EmptyList } from "../components/Search/EmptyList";
+import { useIONavigation } from "../../../navigation/params/AppParamsList";
+import { useIOStore } from "../../../store/hooks";
+import { searchMessagesUncachedSelector } from "../store/reducers/allPaginated";
+import { UIMessage } from "../types";
+import { WrappedMessageListItem } from "../components/Home/WrappedMessageListItem";
+
+const INPUT_PADDING: IOSpacingScale = 16;
+const MIN_QUERY_LENGTH: number = 3;
+
+export const MessagesSearchScreen = () => {
+  const insets = useSafeAreaInsets();
+  const navigation = useIONavigation();
+  const store = useIOStore();
+  const searchInputRef = useRef<SearchInputRef>(null);
+  const [query, setQuery] = useState<string>("");
+  const [filteredMessages, setFilteredMessages] = useState<
+    ReadonlyArray<UIMessage>
+  >([]);
+
+  const renderItemCallback = useCallback(
+    (itemInfo: ListRenderItemInfo<UIMessage>) => (
+      <WrappedMessageListItem index={itemInfo.index} message={itemInfo.item} />
+    ),
+    []
+  );
+  const renderListEmptyComponent = () => {
+    if (query.trim().length < MIN_QUERY_LENGTH) {
+      return (
+        <EmptyList
+          pictogram="searchLens"
+          title={I18n.t("messages.search.emptyState.title")}
+        />
+      );
+    }
+
+    return <VSpacer size={16} />;
+  };
+
+  const handleCancel = useCallback(() => navigation.goBack(), [navigation]);
+
+  useFocusEffect(
+    useCallback(() => {
+      searchInputRef.current?.focus();
+    }, [])
+  );
+
+  useEffect(() => {
+    const timeoutHandleId = setTimeout(() => {
+      const state = store.getState();
+      const searchResult = searchMessagesUncachedSelector(
+        state,
+        query,
+        MIN_QUERY_LENGTH
+      );
+      setFilteredMessages(searchResult);
+    }, 350);
+    return () => clearTimeout(timeoutHandleId);
+  }, [query, setFilteredMessages, store]);
+
+  return (
+    <>
+      <View
+        style={[
+          {
+            marginTop: insets.top,
+            paddingVertical: INPUT_PADDING
+          },
+          IOStyles.horizontalContentPadding
+        ]}
+      >
```

Comment:
```suggestion
  const containerStyle: ViewStyle = useMemo(
    () => ({
      ...IOStyles.horizontalContentPadding,
      marginTop: insets.top,
      paddingVertical: INPUT_PADDING
    }),
    [insets.top]
  );

  return (
    <>
      <View
        style={containerStyle}
      >
```
I believe we should avoid computed styles when possible, especially in a screen with so many renders -- what do you think?  IMO this could help, but I'm not 100% on its real benefits.

Reply:
Yeah, let's apply it